### ISSUE TITLE
Backward incompatible changes introduced in v2.7.0

### ISSUE THREAD

**Issue by f-w** [nodemailer/mailparser#221]
Hi,
My package depends on mailparser@^2.3.2. It used to work. From v2.7.0 onward it breaks when calling simpleParser

```
const simpleParser = require('mailparser').simpleParser;
const msg = `From: Mail Delivery Subsystem <postmaster@gems.invalid.local>
To: <bn-5b50cb6e953d170b24983019-42074@invalid.local>
MIME-Version: 1.0
Content-Type: multipart/report; report-type=delivery-status;
	boundary="w7TItYs4100793.1535568934/foo.invalid.local"
Subject: Returned mail: see transcript for details
Auto-Submitted: auto-generated (failure)

This is a MIME-encapsulated message

--w7TItYs4100793.1535568934/foo.invalid.local
Content-Type: message/delivery-status

Reporting-MTA: dns; foo.invalid.local
Received-From-MTA: DNS; invalid.local
Arrival-Date: Wed, 29 Aug 2018 11:55:34 -0700

Final-Recipient: RFC822; bar@invalid.local
Action: failed
Status: 5.1.1

--w7TItYs4100793.1535568934/foo.invalid.local

`
let parsed = await simpleParser(msg);
```
The call never returns, nor exceptions thrown. Since semver minor version change is supposed to be backward compatible, I believe this is a bug.

**Comment by peernohell** [comment]
Hi @f-w 
I just test your code in a [runkit](https://runkit.com/peernohell/5ab2989d25eebc001213c29d) and simpleParse seems to work correctly.
Could you reproduce ? 
 Otherwise the issue could be just closed.

**Comment by f-w** [comment]
@peernohell , thanks for testing. Upon retry it seems to work so I'll close it for now.

**Comment by f-w** [comment]
The problem is now reproduced in https://repl.it/repls/TartPricklySolution. 

```
`From: Mail Delivery Subsystem <postmaster@gems.invalid.local>
To: <bn-5b50cb6e953d170b24983019-42074@invalid.local>
MIME-Version: 1.0
Content-Type: multipart/report; report-type=delivery-status;
	boundary="w7TItYs4100793.1535568934/foo.invalid.local"
Subject: Returned mail: see transcript for details
Auto-Submitted: auto-generated (failure)

This is a MIME-encapsulated message

--w7TItYs4100793.1535568934/foo.invalid.local
Content-Type: message/delivery-status

Reporting-MTA: dns; foo.invalid.local
Received-From-MTA: DNS; invalid.local
Arrival-Date: Wed, 29 Aug 2018 11:55:34 -0700

Final-Recipient: RFC822; bar@invalid.local
Action: failed
Status: 5.1.1

--w7TItYs4100793.1535568934/foo.invalid.local

`
```

**Comment by andris9** [comment]
This issue is fixed in v2.7.2

Problem was that the provided example email is invalid, it is missing end marker for MIME structure and Mailparser was kept waiting on the finish marker (which never arrived). v2.7.2 stops waiting for missing MIME parts if input stream itself ends.

**Comment by micktaiwan** [comment]
please review #302
